File: ApiBaseCore/ApiBaseCore.Domain/Models/Address.cs
using System.ComponentModel.DataAnnotations;
using System.Text.Json.Serialization;

namespace ApiBaseCore.Domain.Models;

public class Address
{
    [Key]
    public int Id { get; private set; }

    [Required]
    [MaxLength(100)]
    [JsonPropertyName("name")]
    public string Name { get; private set; }

    [Required]
    [MaxLength(200)]
    [JsonPropertyName("street")]
    public string Street { get; private set; }

    [Required]
    [MaxLength(2)]
    [JsonPropertyName("country_code")]
    public string CountryCode { get; private set; }

    public Guid UsuarioId { get; private set; }
    
    [JsonIgnore]
    public Usuario Usuario { get; private set; }

    private Address() { }

    public Address(string name, string street, string countryCode)
    {
        Name = name;
        Street = street;
        CountryCode = countryCode;
    }
}

File: ApiBaseCoreJava/src/main/java/com/polancou/apibasecore/domain/models/Address.java
package com.polancou.apibasecore.domain.models;

import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonProperty;
import jakarta.persistence.*;
import lombok.*;

@Entity
@Table(name = "Addresses")
@Getter
@NoArgsConstructor(access = AccessLevel.PROTECTED)
public class Address {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    @Column(nullable = false, length = 100)
    @JsonProperty("name")
    private String name;

    @Column(nullable = false, length = 200)
    @JsonProperty("street")
    private String street;

    @Column(nullable = false, length = 2)
    @JsonProperty("country_code")
    private String countryCode;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "UsuarioId", nullable = false)
    @JsonIgnore
    private Usuario usuario;

    public Address(String name, String street, String countryCode) {
        this.name = name;
        this.street = street;
        this.countryCode = countryCode;
    }
    
    // Setter for Usuario to be used when adding address to user
    protected void setUsuario(Usuario usuario) {
        this.usuario = usuario;
    }
}
